ixwebsocket/IXProxySocketSetting.cpp

```diff
@@ -0,0 +1,63 @@
+#include "IXProxySocketSetting.h"
+
+#include <iostream>
+
+void ix::ProxySetup::setProxyPass(std::string &proxypass) {
```

Comment:
Still not right :)
Let me explain it in more details.

in C++ there are main types of values: rvalues and lvalues (yes, for "left" and "right" generally).
lvalues are, for example, just variables and class members; rvalues are, for example, some temporary objects (like the function returns something and it's immediately used).

Now let's imagine, that you have a method and inside it you want to save the value to the class member.
If you have `setProxyPass(const std::string& proxypass)  { proxyPass = proxypass; }`
then it will be:
if setProxyPass() called with lvalue as an argument: 1 copy (inside the function)
if setProxyPass() called with rvalue as an argument: 1 copy (inside the function

If you have `setProxyPass(std::string proxypass)  { proxyPass = std::move(proxypass); }`
then it will be
if setProxyPass() called with lvalue as an argument: 1 copy (before the call)
if setProxyPass() called with rvalue as an argument: 0 copies! (temporary object is moved directly to the class member)

As std::string object longer than some length you can google for SSO - short string optimization) are kept in heap and there's no CoW (copy-on-write) in modern C++ standars, each time the object is copied, the memory allocation is performed, and that's a very slow operation (x100-x1000 times slower than just copy some registers or stack-allocated data).

The approach that you have currently in code is the worst one - you pass the argument as a non-const reference and do std::move for it. So for rvalues it's OK, but for lvalues it will implicitly clean the variable on the caller side that may lead to bugs and undefined behavior.

TLDR:
1. If you are using references, always use const references until there's a significant reason to use non-const
2. If you accept an argument and save it to the class member, use pass-by-value (not reference!) + std::move